### ISSUE TITLE
`trim` with a literal 1st arg

### ISSUE THREAD

**Issue by staabm** [vimeo/psalm#5787]
Similar to https://github.com/vimeo/psalm/issues/4070 and https://github.com/vimeo/psalm/issues/3877 it would be useful if psalm could error when trim is used with a literal 1st arg - since this indicates that the code mixed up arg1 and arg2.

Arg2 is usually a literal string, while the 1st arg is always a variable.

**Comment by AndrolGenhald** [comment]
I'm not sure if it's useful in practice, but it's perfectly valid to use a literal as the first argument and a variable as the second: https://3v4l.org/1UtLr

**Comment by staabm** [comment]
> perfectly valid to use a literal as the first argument and a variable as the second:

sure, its valid. but it does not make sense to use it.

**Comment by AndrolGenhald** [comment]
I see it as a bit different than that case. The `substr` issue was that the result was always `false`, and thus could be simplified to just `false` (or more likely fix the incorrect code). In this case it's possible for the `trim` call to be intentional, and it can't be simplified. If _both_ arguments are literals then yes, that could be pointed out as redundant since it could easily just be changed to the resulting literal, but I imagine there could be some usages like this that are intentional and correct:
`echo trim("Some text\n", $trim_newlines ? "\n" : "");`

In practice that usage may be very infrequent, and it's probably usually trivial to refactor it to a better form anyway, so I'm not totally opposed to it since it could help catch mistakes. I think if it is implemented though it might be worth a new issue type, since it's not exactly an error, just a possible mistake (maybe PossiblyReversedArguments or something like that).

It would probably be a good idea to identify other methods that have easily mixed up arguments. Most of them should be handled by type checking (eg all of the array methods), but it might be worth having this check for more than just `trim`.

**Comment by staabm** [comment]
> but it might be worth having this check for more than just trim.

absolutely. I have experienced this very issue today with `trim` therefore the issue was opened specific to this case. There are for sure more cases which could be taken into account.
